src/components/posts/PostLike.js

```diff
@@ -0,0 +1,67 @@
+import like from './like.png';
+import React from 'react';
+import { useState, useEffect } from 'react';
+import client from '../../utils/client';
+import './style.css';
+import colours from '../../utils/colours';
+
+const PostLike = ({likes=[], postId}) => { 
+
+const blue = colours.blue;
+const grey = colours.grey;
+
+const [divStyle, updateStyle] = useState({ backgroundColor: grey });
+const [postLikes,updateLikes] = useState(likes);
+const [id, updateId] = useState('');
+const [talkingToServer, updateTalkStatus] = useState(false);
+
+useEffect(()=>{
+  const like = likes.find(like => String(like.userId) === localStorage.getItem('userId')); // checks if user has liked the post
+  if (like){
+    updateStyle({backgroundColor: blue});
+    updateId(like.id);
+  }
+},[]);
+
+  function likePost(event){
+  updateTalkStatus(true);
+  client.post(`/post/${postId}/like/`)
+  .then(res => {
+    updateId(_ => res.data.data.id);
+    updateLikes(likes => [res.data.data, ...likes]);
+    updateStyle({backgroundColor: blue});
+  })
+  .catch((err) => {
+    console.log({error: `when liking post: ${err.message}`});
+  })
+  .finally(() => updateTalkStatus(false));
+  }
+
+  function unlikePost(event){
```

Comment:
The alignment/formatting of this code looks a bit off, each lines seems to have an extra space in front. Should format this correctly.